ghmirror/core/mirror_requests.py

```diff
@@ -58,6 +58,64 @@ def _cache_response(resp, cache, cache_key):
         cache[cache_key] = resp
 
 
+def _online_request(method, url, data=None,
+                    headers=None, parameters=None):
+    """
+    Handle API errors on conditional requests and try
+    to serve contents from cache
+    """
```

Comment:
I don't see this method doing all that. It just wraps the request conveniently to decide if it's valid or not.